src/Orleans.Serialization.FSharp/FSharpCodecs.cs

```diff
@@ -13,6 +13,31 @@
 
 namespace Orleans.Serialization
 {
+    /// <summary>
+    /// Serializer for <see cref="Unit"/>
+    /// </summary>
+    [RegisterSerializer]
+    public sealed class FSharpUnitCodec : IFieldCodec<Unit>
+    {
+        public void WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, Unit value) where TBufferWriter : IBufferWriter<byte> =>
+            ReferenceCodec.WriteNullReference(ref writer, fieldIdDelta);
+
+        public Unit ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+        {
+            field.EnsureWireType(WireType.Reference);
+            ReferenceCodec.MarkValueField(reader.Session);
+            var reference = reader.ReadVarUInt32();
+            if (reference != 0) throw new UnknownReferencedTypeException(reference);
```

Comment:
To align with what we do for `Nullable<T>`:
```suggestion
            if (reference != 0) throw new ReferenceNotFoundException(typeof(Unit), reference)
```